craft_providers/bases/ubuntu.py

```diff
@@ -47,13 +47,13 @@
 class BuilddBaseAlias(enum.Enum):
     """Mappings for supported buildd images."""
 
-    XENIAL = "16.04"
-    BIONIC = "18.04"
-    FOCAL = "20.04"
-    JAMMY = "22.04"
-    KINETIC = "22.10"
-    LUNAR = "23.04"
-    DEVEL = "devel"
+    XENIAL = ("ubuntu", "16.04")
```

Comment:
Broken record - I'm still not convinced these tuples are adding value.

My concerns:
1. We can already deduce the base more pythonically using the type.  For example, the "ubuntu" and "centos" string aren't needed [here](https://github.com/canonical/craft-providers/pull/249/files#diff-2a629916bf70d9a5b8f644b9d2f37fe36447c8700d46990ab34619c72acd077aR67)
2. I'm not sure it's advantageous to the craft applications.  [This](https://github.com/canonical/charmcraft/pull/1065/files#diff-30a6f50c99bd977857f6dcacd6f4c935f66c39384725962288d623506e07398cR180) is still using type deduction, not the strings.  
3. And finally, although this eliminates the need for charmcraft's mapping `BASE_CHANNEL_TO_PROVIDER_BASE`, the other craft applications will still need a mapping, since they use different naming conventions.
4. The `is_base_available()` function in [charmcraft](https://github.com/canonical/craft-providers/pull/249/files#diff-2a629916bf70d9a5b8f644b9d2f37fe36447c8700d46990ab34619c72acd077aR57) uses the tuple, but that function really belongs in craft-providers. And I think there are better implementations for that validation, such as catching and re-raising the exception from your new `get_base_alias()` function.

Reply:
1. and 2., because I see some place use the type, others use `base.value`. So if we can get every place use type, we don't need these tuples.

3. Maybe we can add theme together? like `core20`

4. OK

Reply:
And `_PROVIDER_BASE_TO_LXD_REMOTE_IMAGE` need to be changed if we want to accept `type` not `tuple` or `str` for mapping.

Reply:
No problem - since it's a private member, feel free to change as needed!